SU2_CFD/src/output/COutput.cpp

```diff
@@ -741,6 +761,16 @@ void COutput::WriteToFile(CConfig *config, CGeometry *geometry, unsigned short f
     delete fileWriter;
 
   }
+
+  if (!config->GetWrt_Restart_Overwrite() && (rank == MASTER_NODE)){
+    /*--- copy the file with iteration number appended ---*/
+    std::ifstream  src(restartFilename.c_str(), std::ios::binary);
+    std::ofstream  dst(filename_iter.c_str(),   std::ios::binary);
+    dst << src.rdbuf();
+
+  }
```

Comment:
Change the file writer so that you can pass the filename to Write_Data(), and call it twice.
Most of the writers are parallel, here you are using a single core to do the copy.

Reply:
Yes, it should be parallel, but it's less intrusive to just instantiate a new CFileWriter then.

Reply:
I will not approve solution that involve duplicating patterns.
If you delete the "fileName" member variable from the writer, you immediately see where it is being used, and so the chance of that introducing a bug is very low.

Reply:
OK, I changed the file writer, and also switched to enum class for the enum_output while we're at it. The filename is not part of the constructor but an argument of Write_Data.